src/libaktualizr/uptane/secondaryinterface.h

```diff
@@ -26,15 +33,16 @@ class SecondaryInterface {
   virtual int32_t getRootVersion(bool director) const = 0;
   virtual bool putRoot(const std::string& root, bool director) = 0;
 
-  virtual bool sendFirmware(const std::string& data) = 0;
-  virtual data::ResultCode::Numeric install(const std::string& target_name) = 0;
+  virtual data::ResultCode::Numeric install(const Target& target) = 0;
 
   virtual ~SecondaryInterface() = default;
 
  public:
-  // make children non-copyable
+  // make children non-copyable and non-movable
```

Comment:
I think I've just put all that 'non-copyable' and 'non-movable' stuff there because that's what is usually is required for a base class to avoid the slicing problem. But in this case, it's an abstract class, so it doesn't make sense to keep all that stuff in the interface declaration and let specific non-abstract children decide whether to make a class non-copyable/non-movable or not in their specific case. So I am just removing all that stuff.